packages/runtime/src/runtime/screen/index.tsx

```diff
@@ -49,8 +51,23 @@ export const EnsembleScreen: React.FC<EnsembleScreenProps> = ({
     inputs,
   ) as { [key: string]: unknown };
 
+  const processedAPIs = processApiDefinitions(screen.apis);
```

Comment:
This should be in useEffect? This should only happen once per render. Make this a hook.